check_brevisone

```diff
@@ -177,21 +188,104 @@ def get_data(base_url, timeout, insecure):
 
     return resp
 
-def extract_perfdata(lines):
+def parse_data(data):
     """
-    Safely extract perfdata
+    Safely extract data from the APIs reponse
     """
-    if len(lines) < 4:
-        return None
+    lines = [str(i).strip() for i in data.split("\n") if i]
+
+    parsed_data = {}
+
+    for l in lines:
+        d = l.split(":")
+        if len(d) == 2:
+            key = d[0].strip()
+            value = d[1].strip()
+
+            # Remove the db string from the signal value
+            if key == "signal":
+                value = value.replace('db', '').replace('dBm', '').strip()
 
-    perfdata = {
-        'que': read_int(lines[0]),
-        'failed': read_int(lines[1]),
-        'signal': read_signal(lines[2]),
-        'total': read_int(lines[3])
-    }
+            # Parse integer value to be actual integers
+            if value.lstrip('-').isdigit():
+                value = int(value)
 
-    return perfdata
+            parsed_data[key] = value
+
+    return parsed_data
+
+
+def worst_state(*states):
+    overall = -1
+
+    for state in states:
+        if state == CRITICAL:
+            overall = CRITICAL
+        elif state == UNKNOWN:
+            if overall != CRITICAL:
+                overall = UNKNOWN
+        elif state > overall:
+            overall = state
+
+    if overall < 0 or overall > 3:
+        overall = UNKNOWN
+
+    return overall
+
+def debug_print(debug_flag, message):
+    """
+    Print debug messages if -d is set.
+    """
+    if not debug_flag:
+        return
+
+    print(message)
+
+def determine_status(args, perfdata):
+    states = []
+    outputs = []
+
+    if perfdata['failed'] >= args.failed_critical:
+        outputs.append(" \\_[CRITICAL] Failed sending: {}".format(perfdata['failed']))
+        states.append(CRITICAL)
+    elif perfdata['failed'] >= args.failed_warning:
+        outputs.append(" \\_[WARNING] Failed sending: {}".format(perfdata['failed']))
+        states.append(WARNING)
+    else:
+        outputs.append(" \\_[OK] Failed sending: {}".format(perfdata['failed']))
+        states.append(OK)
+
+    if perfdata['signal'] <= args.signal_critical:
+        outputs.append(" \\_[CRITICAL] Signal strength: {}".format(perfdata['signal']))
+        states.append(CRITICAL)
+    elif perfdata['signal'] <= args.signal_warning:
+        outputs.append(" \\_[WARNING] Signal strength: {}".format(perfdata['signal']))
+        states.append(WARNING)
+    else:
+        outputs.append(" \\_[OK] Signal strength: {}".format(perfdata['signal']))
+        states.append(OK)
+
+    if perfdata['que'] >= args.queue_critical:
+        outputs.append(" \\_[CRITICAL] Que length: {}".format(perfdata['que']))
```

Comment:
They might short *queue* to *que* in the API for some reason (save two bytes?!?), but I would prefer the long version in the output (and probably in the perfdata too).

Reply:
sorry for the delay btw.

Reply:
Done